fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1007,9 +1008,11 @@ public void buildIndex() {
     /**
      * Wait for asynchronous index build to complete.
      * @param buildIndexFuture the result of {@link #buildIndexAsync}
+     * @param <T> return type of function to run
+     * @return future that will contain the result of {@code retriable} after successful run and commit
      */
-    public void asyncToSync(@Nonnull CompletableFuture<Void> buildIndexFuture) {
-        runner.asyncToSync(FDBStoreTimer.Waits.WAIT_ONLINE_BUILD_INDEX, buildIndexFuture);
+    public <T> T asyncToSync(@Nonnull CompletableFuture<T> buildIndexFuture) {
```

Comment:
It's weird to me that this is a public method, but this is probably not the right PR to deal with that.

Reply:
Is this being modified to make the unit test easier?

Reply:
- Yes, I think it make sense not exposing `asyncToSync` but it is a breaking change.

- Yes, it was modified to make the unit test easier.

Reply:
I think we could probably mark it either `INTERNAL` or `UNSTABLE` and then think about removing it (i.e., making it...package private?) in a future release.